src/app/functions/server/onerep.ts

```diff
@@ -439,3 +449,47 @@ export async function fetchAllPages<Data>(
 
   return dataList.flat();
 }
+
+// Local instance map to cache results to prevent excessive API requests
+// Would be nice to share this cache with other pod via Redis in the future
+const profileStatsCache = new Map<string, ProfileStats>();
+export async function getProfilesStats(
+  from: Date,
```

Comment:
For [MNTOR-2315](https://mozilla-hub.atlassian.net/browse/MNTOR-2315) we need the total number of scans performed and I guess we could just use this function as well. We could make the `from` prop optional as well and in case no time period is provided just return the total — or do you prefer us explicitly passing `from`.

Reply:
@flozia ah, that makes sense. I can make the `from` optional as well

Reply:
Fixed